gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -0,0 +1,150 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.common.collect.ImmutableList;
+import io.grpc.CallOptions;
+import io.grpc.ClientCall;
+import io.grpc.ManagedChannel;
+import io.grpc.MethodDescriptor;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A {@link ManagedChannel} that will send requests round robin via a set of channels.
+ *
+ * <p>Package-private for internal use.
+ */
+class ChannelPool extends ManagedChannel {
+  private final ImmutableList<ManagedChannel> channels;
+  private final AtomicInteger requestCount = new AtomicInteger();
+  private final String authority;
+
+  /**
+   * Initializes the channel pool. Assumes that all channels have the same authority.
+   *
+   * @param channels a List of channels to pool.
+   */
+  ChannelPool(List<ManagedChannel> channels) {
+    this.channels = ImmutableList.copyOf(channels);
+    authority = channels.get(0).authority();
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public String authority() {
+    return authority;
+  }
+
+  /**
+   * Create a {@link ClientCall} on a Channel from the pool chosen in a round-robin fashion to the
+   * remote operation specified by the given {@link MethodDescriptor}. The returned {@link
+   * ClientCall} does not trigger any remote behavior until {@link
+   * ClientCall#start(ClientCall.Listener, io.grpc.Metadata)} is invoked.
+   */
+  @Override
+  public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(
+      MethodDescriptor<ReqT, RespT> methodDescriptor, CallOptions callOptions) {
+
+    return getNextChannel().newCall(methodDescriptor, callOptions);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public ManagedChannel shutdown() {
+    for (ManagedChannel channelWrapper : channels) {
+      channelWrapper.shutdown();
+    }
+
+    return this;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public boolean isShutdown() {
+    for (ManagedChannel channel : channels) {
+      if (!channel.isShutdown()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public boolean isTerminated() {
+    for (ManagedChannel channel : channels) {
+      if (!channel.isTerminated()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public ManagedChannel shutdownNow() {
+    for (ManagedChannel channel : channels) {
+      channel.shutdownNow();
+    }
+    return this;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
+    long endTimeNanos = System.nanoTime() + unit.toNanos(timeout);
+    for (ManagedChannel channel : channels) {
+      if (channel.isTerminated()) {
+        continue;
+      }
```

Comment:
I don't think we need to do this right? if the channel is already terminated, then `awaitTermination` will return immediately.

Reply:
Yep, removed